### DISCUSSION TITLE
RandomLinkSplit with `rev_edge_types` and `undirected=True`

### DISCUSSION THREAD

**Discussion by denadai2** [pyg-team/pytorch_geometric#7980] [Q&A] [answered]
Dear community,
I am using `RandomLinkSplit` with `rev_edge_types` ('book', 'rev_co-read', 'book'), and `undirected=True`. I do not understand what's happening at this point of the code: https://github.com/pyg-team/pytorch_geometric/blob/60a2948ba1745e35ee5209d5e3d0b861aca54495/torch_geometric/transforms/random_link_split.py#L273

Why do we do so, if we have rev_edges in hetero graphs? Why is it needed? 

thank you!

**Comment by rusty1s** [comment, marked as answer]
Mh, I agree. It is a bit misleading. Currently, the `undirected=True` option assumes that a single edge type holds bidirectional edges. As such, we `undirected=False` internally whenever the edge type models a bipartite graph, or whenever `edge_type != rev_edge_type`.

**Comment by denadai2** [comment]
thanks!
